src/main/java/com/bean00/RequestReader.java

```diff
@@ -0,0 +1,39 @@
+package com.bean00;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class RequestReader {
+    private BufferedReader in;
+
+    public RequestReader(BufferedReader in) {
+        this.in = in;
+    }
+
+    public String getRequestLine() throws IOException {
+        return in.readLine();
+    }
+
+    public ArrayList<String> getHeaders() throws IOException {
+        ArrayList<String> headers = new ArrayList<>();
+        String header;
```

Comment:
I think it's better to declare this on line 23 inside the while loop if you don't need it outside of the while loop!

Reply:
So I tried doing that (change it to `String header = in.readLine();` as the first statement inside the while loop), but there was an error (`Error: java: cannot find symbol symbol: variable header`). I believe the `header` variable still needs to be declared outside of the loop, because the check for the while is outside the loop.

Reply:
ohh woops, I did not see that it's being used there. My bad. I think you can put line 23 inside the while loop too actually, but upto you!